onnxruntime/core/providers/cpu/containers.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(USE_MIMALLOC_STL_ALLOCATOR)
+#include <mimalloc.h>
+#endif
+
+namespace onnxruntime {
+
+template <class T>
+struct BFCArenaAllocator {
+  AllocatorPtr arena;
```

Comment:
This class should have some documentation explaining what it is as it's probably not obvious to everyone else. 

I'd also suggest moving it to something like framework/bfc_arena_allocator.h so that this file just defines the containers that use it or the mimalloc allocator.